File: 1737-MaximumNestingDepthOfTheParentheses/1737-MaximumNestingDepthOfTheParentheses.py
# Last updated: 07/04/2026, 11:36:03
class Solution:
    def maxDepth(self, s: str) -> int:
        st = []
        m = 0 
        for i in s :
            if  i == "(" :
                st.append(i) 
            elif i == ")":
                st.pop()
            m = max(len(st),m)

        return m
        

File: 1737-MaximumNestingDepthOfTheParentheses/1737-MaximumNestingDepthOfTheParentheses.cpp
// Last updated: 07/04/2026, 11:35:57
class Solution {
public:
    int maxDepth(string s) {
        int ans = 0 ;
        int bal = 0;

        for ( char c :s){
            if ( c =='('){
                bal++;
            }else if ( c ==')'){
                bal--;
            }
            ans = max(bal,ans);
        }
        //ans = max(bal,ans);
        return ans;
    }
};